modules/charon-core/src/test/java/org/wso2/charon3/core/utils/PatchOperationUtilTest.java

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.charon3.core.utils;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.charon3.core.attributes.Attribute;
+import org.wso2.charon3.core.attributes.ComplexAttribute;
+import org.wso2.charon3.core.attributes.MultiValuedAttribute;
+import org.wso2.charon3.core.attributes.SimpleAttribute;
+import org.wso2.charon3.core.encoder.JSONDecoder;
+import org.wso2.charon3.core.exceptions.BadRequestException;
+import org.wso2.charon3.core.exceptions.CharonException;
+import org.wso2.charon3.core.exceptions.InternalErrorException;
+import org.wso2.charon3.core.exceptions.NotImplementedException;
+import org.wso2.charon3.core.objects.AbstractSCIMObject;
+import org.wso2.charon3.core.objects.User;
+import org.wso2.charon3.core.schema.AttributeSchema;
+import org.wso2.charon3.core.schema.SCIMAttributeSchema;
+import org.wso2.charon3.core.schema.SCIMDefinitions;
+import org.wso2.charon3.core.schema.SCIMResourceTypeSchema;
+import org.wso2.charon3.core.utils.codeutils.PatchOperation;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.charon3.core.schema.SCIMDefinitions.DataType.COMPLEX;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.DataType.STRING;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.Mutability.READ_WRITE;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.Returned.DEFAULT;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.Uniqueness.NONE;
+
+/**
+ * Test class of PatchOperationUtil.
+ */
+public class PatchOperationUtilTest {
+
+    final String remove = "remove";
+    final String add = "add";
+    final String replace = "replace";
+
+    @DataProvider(name = "dataForPatchRemoveSuccess")
+    public Object[][] dataToPatchRemoveSuccess() throws InstantiationException, IllegalAccessException,
+            CharonException {
+
+        PatchOperation patchOperation1 = PatchOperation.class.newInstance();
```

Comment:
Why have you used reflection, while you can easily use the constructor to create the object?

Reply:
fixed in 0700172411ecaa602656c20ddb85d0a7b4f4f2be